src/merkle/smt/full/tests.rs

```diff
@@ -412,21 +414,49 @@ fn test_prospective_insertion() {
 
     let mutations = smt.compute_mutations(vec![(key_1, value_1)]);
     assert_eq!(mutations.root(), root_1, "prospective root 1 did not match actual root 1");
-    smt.apply_mutations(mutations).unwrap();
+    let revert = smt.apply_mutations_with_reversion(mutations).unwrap();
```

Comment:
With this change (and other changes in this file) we are no longer testing the `apply_mutations()` method by itself. Since these are implemented as two different methods, we should probably test both? Or maybe just the newly added tests should be enough?

Reply:
Yes, it's better to tests both, thank you for noticing!